state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -636,3 +636,23 @@ fn test_refund_fee_calculation() {
 		assert_eq!(Swapping::take_refund_fee(3, Asset::Eth), Ok(0));
 	});
 }
+
+#[test]
+fn gas_calculation_can_handle_invalid_swap_rate() {
+	new_test_ext().execute_with(|| {
+		fn test_invalid_swap_rate(swap_rate: f64) {
+			SwapRate::set(swap_rate);
+			assert_eq!(
+				Swapping::calculate_input_for_gas_output::<Ethereum>(
```

Comment:
I think this is a reasonable test, but it does not test the panic case I mentioned above. Reverting the change, this test still passes. This is because in all these cases, we get a None from `calculate_input_for_desired_output`. But we want to get a value that is > u64::MAX and then try to convert it to u64